### ISSUE TITLE
nvs_flash_init_partition_ptr() calls init_custom() in NVSPartitionManager with wrong arguments (IDFGH-7503)

### ISSUE THREAD

**Issue by sascha90024** [espressif/esp-idf#9070]
## Environment

Code snippets in the description of the problem are copied from github from "Latest commit [4dd8832](https://github.com/espressif/esp-idf/commit/4dd88329c1e8b6fb3afc0355b1d5fad0d7a624e9) on Jun 2, 2021"

## Problem Description

The line 53 in nvs_partition_manager.cpp calls init_custom() passing the second argument (sector number) counting from the beginning of the partition:

```cpp
result = init_custom(p, 0, size / SPI_FLASH_SEC_SIZE);
```

The line 115 in nvs_api.cpp invokes init_custom() and passes the second argument (sector number) counting from the beginning of the flash address space:

```cpp
esp_err_t init_res = NVSPartitionManager::get_instance()->init_custom(part,
            partition->address / SPI_FLASH_SEC_SIZE,
            partition->size / SPI_FLASH_SEC_SIZE);
```

but it should be

```cpp
esp_err_t init_res = NVSPartitionManager::get_instance()->init_custom(part,
            0,
            partition->size / SPI_FLASH_SEC_SIZE);
```

### Actual Behavior

Calling nvs_flash_init_partition_ptr(const esp_partition_t *partition) will end up passing the absolute address of the partition divided by SPI_FLASH_SEC_SIZE as the second argument to init_custom().
Calling nvs_flash_init_partition(const char * name) for the same partition will end up passing 0 as the second argument to init custom().

**Comment by igrr** [comment]
Thanks for reporting this! Looks like https://github.com/espressif/esp-idf/pull/8755 can be merged to fix this issue.

**Comment by Alvin1Zhang** [comment]
Thanks for reporting, fix on master branch is available at https://github.com/espressif/esp-idf/commit/87180907f740c405ca8c14b9cdbac987bd52a16e. We will also backport the fix to release/4.4 and release/4.3, thanks.
